source/coffee/image-picker.coffee

```diff
@@ -179,7 +189,46 @@ class ImagePickerOption
     imgAlt = @option.data("img-alt")
     if imgAlt
       image.attr('alt', imgAlt);
-    thumbnail.on("click", @clicked)  
+    thumbnail.on("click", @clicked)
+    thumbnail.on("keydown", (event) ->
+      if event.which == 0 || event.which == 32
+        event.preventDefault()
+        thumbnail.click()
+      else if event.which == 37 || event.which == 40
+        event.preventDefault();
+        $(this).parent().prev().find(".thumbnail").prop("tabindex", "0");
+        $(this).parent().prev().find(".thumbnail").focus();
+      else if event.which == 39 || event.which == 40
+        event.preventDefault();
+        $(this).parent().next().find(".thumbnail").prop("tabindex", "0");
+        $(this).parent().next().find(".thumbnail").focus();
+      return
+    )
+    thumbnail.on("focusout", (event) ->
+      exitingCtrl = !$(this).siblings().is($(event.relatedTarget).closest("li"));
+      $(this).closest("ul").find(".thumbnail").each () ->
+        if $(this).hasClass("selected") && exitingCtrl
+          $(this).prop("tabindex", "0");
+        else
+          $(this).prop("tabindex", "-1");
+        return
+      if $(this).closest("ul").find(".selected").length == 0 && exitingCtrl
+        $("li:first-child", $(this).closest("ul")).find(".thumbnail").prop("tabindex", "0");
+      return
```

Comment:
This looks pretty similar to the code above, should it be refactored?

Reply:
Will look into refactoring this